tests/base/test_regularization.py

```diff
@@ -782,9 +783,12 @@ def test_active_cells(self, mesh, regularization_class):
     def test_weights(self, mesh):
         """Test cell_weights and weights."""
         weights = np.ones(len(mesh))
+        weights_dict = {"weights": weights}
         msg = "Cannot simultaneously pass 'weights' and 'cell_weights'."
         with pytest.raises(ValueError, match=msg):
-            BaseRegularization(mesh, weights=weights, cell_weights=weights)
+            BaseRegularization(
+                mesh, weights=weights_dict.weights, cell_weights=weights_dict.weights
+            )
```

Comment:
When running the tests, Python complains because `weights_dict` is a dictionary, and dictionaries don't have a `weights` attribute. What you actually want to do is to the get the value under the `"weights"` key. This can be done with `weights_dict["weights"]`.

I noticed there are a few other places in your PR where you will need to apply a similar fix.

Let me know if you need any help with that!
```suggestion
            BaseRegularization(
                mesh, weights=weights_dict["weights"], cell_weights=weights_dict["weights"]
            )
```